### ISSUE TITLE
java.lang.IllegalArgumentException: Cannot create a VALUES() constructor with an empty set of rows when using JSON_ARRAY(ABSENT ON NULL) in PostgreSQL

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#13054]
When emulating `JSON_ARRAY(ABSENT ON NULL)` (i.e. an empty array) on PostgreSQL, the following exception is raised:

```
Exception in thread "main" java.lang.IllegalArgumentException: Cannot create a VALUES() constructor with an empty set of rows
	at org.jooq/org.jooq.impl.Values.assertNotEmpty(Values.java:112)
	at org.jooq/org.jooq.impl.Values.<init>(Values.java:107)
	at org.jooq/org.jooq.impl.DSL.values(DSL.java:30708)
	at org.jooq/org.jooq.impl.JSONArray.accept(JSONArray.java:150)
	at org.jooq/org.jooq.impl.DefaultRenderContext.visit0(DefaultRenderContext.java:726)
	at org.jooq/org.jooq.impl.AbstractContext.visit(AbstractContext.java:246)
	at org.jooq/org.jooq.impl.QueryPartCollectionView.acceptElement(QueryPartCollectionView.java:243)
	at org.jooq/org.jooq.impl.QueryPartCollectionView.accept(QueryPartCollectionView.java:221)
	at org.jooq/org.jooq.impl.DefaultRenderContext.visit0(DefaultRenderContext.java:726)
	at org.jooq/org.jooq.impl.AbstractContext.visit(AbstractContext.java:295)
	at org.jooq/org.jooq.impl.SelectQueryImpl.toSQLReference0(SelectQueryImpl.java:2193)
	at org.jooq/org.jooq.impl.SelectQueryImpl.lambda$33(SelectQueryImpl.java:1857)
	at org.jooq/org.jooq.impl.AbstractContext.toggle(AbstractContext.java:338)
	at org.jooq/org.jooq.impl.AbstractContext.data(AbstractContext.java:349)
	at org.jooq/org.jooq.impl.SelectQueryImpl.toSQLReferenceLimitDefault(SelectQueryImpl.java:1857)
	at org.jooq/org.jooq.impl.SelectQueryImpl.accept0(SelectQueryImpl.java:1771)
	at org.jooq/org.jooq.impl.SelectQueryImpl.accept(SelectQueryImpl.java:1426)
	at org.jooq/org.jooq.impl.DefaultRenderContext.visit0(DefaultRenderContext.java:726)
	at org.jooq/org.jooq.impl.AbstractContext.visit(AbstractContext.java:295)
	at org.jooq/org.jooq.impl.QueriesImpl.accept(QueriesImpl.java:168)
	at org.jooq/org.jooq.impl.DefaultRenderContext.visit0(DefaultRenderContext.java:726)
	at org.jooq/org.jooq.impl.AbstractContext.visit(AbstractContext.java:295)
	at org.jooq/org.jooq.impl.DefaultDSLContext.render(DefaultDSLContext.java:709)
	at org.jooq/org.jooq.ParserCLI.render(ParserCLI.java:493)
	at org.jooq/org.jooq.ParserCLI.interactiveMode(ParserCLI.java:341)
	at org.jooq/org.jooq.ParserCLI.lambda$0(ParserCLI.java:82)
	at org.jooq/org.jooq.CLIUtil.main(CLIUtil.java:55)
	at org.jooq/org.jooq.ParserCLI.main(ParserCLI.java:72)
```

This bug has been found thanks to a new test coverage improvement from: https://github.com/jOOQ/jOOQ/issues/13051

**Comment by lukaseder** [comment]
CockroachDB and Yugabyte are also affected

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.17.0, 3.16.5 (https://github.com/jOOQ/jOOQ/issues/13055), and 3.15.9 (https://github.com/jOOQ/jOOQ/issues/13056)
